static/compiler.js

```diff
@@ -106,47 +150,69 @@ function Compiler(domRoot, origFilters, windowLocalPrefix, onChangeCallback, lan
             break;
     }
 
+    // Set up the editor's window
+    // TODO : understand why [0] is mandatory here
     cppEditor = CodeMirror.fromTextArea(domRoot.find(".editor textarea")[0], {
         lineNumbers: true,
         matchBrackets: true,
         useCPP: true,
         mode: cmMode
     });
+
     // With reference to "fix typing '#' in emacs mode"
     // https://github.com/mattgodbolt/gcc-explorer/pull/131
     cppEditor.setOption("extraKeys", {
       "Alt-F": false
     });
     cppEditor.on("change", function () {
         if ($('.autocompile').hasClass('active')) {
-            onChange();
+            onEditorChange();
         }
     });
-    asmCodeMirror = CodeMirror.fromTextArea(domRoot.find(".asm textarea")[0], {
-        lineNumbers: true,
-        mode: "text/x-asm",
-        readOnly: true,
-        gutters: ['CodeMirror-linenumbers']
-    });
 
+    // Load/Save/Remove setting from the browser
     function getSetting(name) {
+        // source: http://stackoverflow.com/questions/9943220/how-to-delete-a-localstorage-item-when-the-browser-window-tab-is-closed
         return window.localStorage[windowLocalPrefix + "." + name];
     }
-
     function setSetting(name, value) {
         window.localStorage[windowLocalPrefix + "." + name] = value;
     }
+    function removeSetting(name) {
+        window.localStorage.removeItem(windowLocalPrefix + "." + name);
+    }
+
 
     var codeText = getSetting('code');
     if (!codeText) codeText = $(".template.lang." + lang.replace(/[^a-zA-Z]/g, '').toLowerCase()).text();
     if (codeText) cppEditor.setValue(codeText);
-    domRoot.find('.compiler_options').change(onChange).keyup(onChange);
+
     ignoreChanges = false;
 
-    if (getSetting('compilerOptions')) {
-        domRoot.find('.compiler_options').val(getSetting('compilerOptions'));
+
+
+    // function similar to setCompilersInSlot
+    // TODO : should be returned by Compiler() !
+    // (not executed in setCompilers)
+    function setLeaderSlotMenu() {
```

Comment:
In general I'm not sure about the whole "leader slot" thing. I wonder if there might be a better way of doing this, though I'm a little short on ideas now. I find it rather unintuitive from a UI point of view.